synapse/storage/databases/main/task_scheduler.py

```diff
@@ -38,13 +40,18 @@ def __init__(
         super().__init__(database, db_conn, hs)
 
     @staticmethod
-    def _convert_row_to_task(row: Dict[str, Any]) -> ScheduledTask:
-        row["status"] = TaskStatus(row["status"])
-        if row["params"] is not None:
-            row["params"] = db_to_json(row["params"])
-        if row["result"] is not None:
-            row["result"] = db_to_json(row["result"])
-        return ScheduledTask(**row)
+    def _convert_row_to_task(row: ScheduledTaskRow) -> ScheduledTask:
+        task_id, action, status, timestamp, resource_id, params, result, error = row
```

Comment:
Alternately we could just do `row[0]`, etc.

Reply:
Those help readability, but from a quick test it would allocate another list:

```
>>> first, *rest = (1, 2, 3, 4)
>>> first
1
>>> rest
[2, 3, 4]
>>> first, *_ = (1, 2, 3, 4)
>>> _
[2, 3, 4]
```

Reply:
Sorry if I'm unclear, I mean below this line of code we could do 

```python
ScheduledTaskRow(
    task_id=row[0],
    action=row[1],
    ...
)
```

Reply:
Oh, I see. No strong opinions here.